controllers/automation/policyautomation_controller.go

```diff
@@ -199,6 +199,105 @@ func (r *PolicyAutomationReconciler) Reconcile(
 			} else {
 				log.Info("All clusters are compliant. Doing nothing.")
 			}
+		} else if policyAutomation.Spec.Mode == "everyEvent" {
+			log := log.WithValues("mode", "everyEvent")
+			targetList := common.FindNonCompliantClustersForPolicy(policy)
```

Comment:
You need to filter out the clusters in `targetList` that are also in `policyAutomation.Status.ClustersWithEvent` since `FindNonCompliantClustersForPolicy` just gives you the current noncompliant clusters but the last time the `Reconcile` method ran, those clusters could have already been noncompliant.

Reply:
I trimmed ClustersWithEvent first and then filter `targetList` as trimmedTargetList on line 275